db.go

```diff
@@ -352,19 +357,90 @@ type DB struct {
 	mu                 sync.RWMutex
 	models             map[reflect.Type]*Model
 	mappings           map[reflect.Type]fieldMap
+
+	// Function to add a deadline to a query.
+	deadlineQueryRewriter func(db *DB, query interface{}, millis int64) (queryWithDeadline interface{}, err error)
+}
+
+type DBOption func(*DB)
+
+var (
+	// When passed as an option to NewDB(), disables query deadlines.
+	QueryDeadlineNone = func(db *DB) {
```

Comment:
Technically, this would allow another package to replace the definition of these methods. Any particular reason to want function variables here instead of functions? E.g.:
```
func QueryDeadlineNone(db *DB) { ... }
```